class.jetpack.php

```diff
@@ -4104,6 +4105,14 @@ function build_connect_url( $raw = false, $redirect = false, $from = false, $reg
 
 		if ( $register || ! $token || ! $site_id ) {
 			$url = Jetpack::nonce_url_no_esc( Jetpack::admin_url( 'action=register' ), 'jetpack-register' );
+
+			if ( ! empty( $redirect ) ) {
```

Comment:
The main reason I didn't do that is because there's a place in the else block that re-uses the original `$redirect` parameter - on [line 4120](https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L4120). I thought it would be easier to read if the $redirect URL was dealt with separtely in each case, than if there were two redirect variables (the original needed for line 4120 and the new one needed in other places) defined from the start.